### DISCUSSION TITLE
Thousands of tenets or just 1?

### DISCUSSION THREAD

**Discussion by fairwindtech** [fief-dev/fief#468] [Q&A]
I am working on an app that will allow will essentially have two types of users: "public" and "business". 

For public, I will be using a single tenant with open registration using OAuth.

For businesses, I will be using email/password without registration enabled. I envision 3 roles for business users - owners, managers, and employees. Owners can be over multiple locations, managers manage a single location, and employees are generally at a single location.

My idea is to create a separate tenet for each location with the owner an admin of each tenent. If John owns a business with two locations, his email will be an admin of two tenants. Then John can then use the fief admin tool to add/manage managers to each location in the manager role, and those managers can use the same tool to add/manage employees in that location. I am hoping to offload all this user management to the businesses and not have to develop a custom user management solution with the API since one already exists in Fief.

From my apps perspective, when a user signs in it just needs to know what location(s) the user is associated with and what roles they are in for those locations, to provide different functionality to the user types for a specific location.

My research has shown two different opinions on the approach:
- The way I am envisioning it creating tenets for each location and turning the owners loose to manage employees via fief admin
- Using a single tenant for all users and segmenting them in my app by associating them with locations and roles in my database and creating my own user management tools for owners. I would like to avoid reinventing the wheel for user management in my app.

My questions are:
- Is multi-tenent approach the best way to handle my use case? 
- If not, what is better and why?
- if so, can fief support many thousand of tenants?

**Comment by fief-bailiff** [comment]
Hail, @fairwindtech 👋 Welcome to Fief's kingdom!

Our team will get back to you very soon to help.

In the meantime, take a minute to [star our repository](https://github.com/fief-dev/fief) ⭐️

<p align="center">
<img width="656" alt="star-fief" src="https://github.com/fief-dev/fief/assets/1144727/b2db7f59-102f-47e5-9751-67fa10372c52">
</p>

## Want to support us?

Subscribe to one of our paid plan to help us continue our work and receive exclusive information and benefits! Starts at $5/month 🪙

<p align="center">
    <a href="https://polar.sh/fief-dev">
    <picture>
      <source media="(prefers-color-scheme: dark)" srcset="https://polar.sh/embed/subscribe.svg?org=fief-dev&darkmode=1">
      <img alt="Subscribe" src="https://polar.sh/embed/subscribe.svg?org=fief-dev">
    </picture>
    </a>
</p>

Farewell!

**Comment by frankie567** [comment]
From what I understand, your use case is closer to user groups/organizations management, which I recommend to implement yourself in your app, since you look to have quite specific requirements.

The problem you would have with tenants is not their number but rather how you'll integrate them in your app. It's indeed worth to note that each tenant have their own Clients, so you would need to have pairs of client ID/secret stored somewhere and instantiate a Fief client accordingly, which could be a bit complex for a lot of tenants.
